onnxruntime/core/providers/cpu/tensor/scatter_nd.cc

```diff
@@ -150,13 +149,16 @@ Status ScatterNDBase::PrepareForCompute(OpKernelContext* context, Prepare& p) co
   for (int64_t i = 0; i < offset_count; ++i) {
     for (int64_t j = 0; j < last_indice_dimension; ++j) {
       auto indice = *(indice_offset + i * last_indice_dimension + j);
-      if (indice < 0 || indice >= input_shape[j]) {
-        err_indice = indice;
+      if (indice < -input_shape[j] || indice >= input_shape[j]) {
```

Comment:
If `indice >= 0`, then indice cannot be `< -input_shape[j]`. 3 tests are computed, if `indice < 0` is done first, only 2 are needed in all cases.

Reply:
ah... this is very subtle... I wonder if it will make too much of a difference in real world, but I rewrote according to your suggestion. :)